File: simpleCity-App/simpleCity-app-frontend/src/app/common/cart-item-item.ts

export class CartItemItem {

//     imageUrl: string;
//     quantity: number;
//     unitPrice: number;
//     productId: string; 

//     constructor(imageUrl: string, quantity: number, unitPrice: number, productId: string) {
//         this.imageUrl = imageUrl;
//         this.quantity = quantity;
//         this.unitPrice = unitPrice;
//         this.productId = productId;
//    }

    constructor(
        public imageUrl: string = '',
        public unitPrice: number = 0,
        public quantity: number = 0,
        public productId: string = '') { }
}

File: app/src/main/java/com/simple_city/app/entities/CartItemItem.java
package com.simple_city.app.entities;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.springframework.web.bind.annotation.CrossOrigin;

@Entity
@Getter
@Setter
@Table(name = "cart_items")
@CrossOrigin(origins = "http://localhost:4200")
public class CartItemItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "quantity")
    private int quantity;

    @Column(name = "unit_price")
    private int unitPrice;

//    @ManyToOne
//    @JoinColumn(name = "product_id", nullable = false)
//    private Product product;
    @Column(name = "product_id")   //@TODO check if product id is cast as a different type in the project
    private Long productId;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "cart_id")
    private Cart cart;

    public void setCustomer(Customer savedCustomer) {
    }
}


